pytorch_lightning/trainer/trainer.py

```diff
@@ -576,37 +576,32 @@ def __init__(
 
     def _init_debugging_flags(
         self,
-        limit_train_batches,
-        limit_val_batches,
-        limit_test_batches,
-        limit_predict_batches,
-        val_check_interval,
-        overfit_batches,
-        fast_dev_run,
-    ):
-        if not isinstance(fast_dev_run, (bool, int)):
-            raise MisconfigurationException(
-                f"fast_dev_run={fast_dev_run} is not a valid configuration. It should be either a bool or an int >= 0"
-            )
```

Comment:
is this handled somewhere else already?

Reply:
My blind guess is that Danielle removed it because `isinstance` checks go against Python's duck-typing philosophy.

We don't check that the input argument is the expected type in 99% of the cases (unless we need to validate a value, as right below).

That's mypy's job.

Reply:
yes, exactly! thanks @carmocca